cogito/protocol.go

```diff
@@ -162,9 +162,14 @@ type Source struct {
 	//
 	// Mandatory
 	//
-	Owner       string `json:"owner"`
-	Repo        string `json:"repo"`
+	Owner string `json:"owner"`
+	Repo  string `json:"repo"`
+	// Mandatory if not using github app auth
 	AccessToken string `json:"access_token"` // SENSITIVE
+
+	// Mandatory if using github app auth
+	GitHubApp github.GitHubApp `json:"github_app,omitempty"`
```

Comment:
if tried with pointers originally: `GitHubApp *github.GitHubApp `json:"github_app,omitempty"`` which made the following more natural:

 `if src.GitHubApp != nil ` but this brings other set of issues with panics due to nil pointer... Easily solved, but not worth it in my opinion...

Reply:
Note that `omitempty` and the newer `omitzero` are used on encoding, not decoding.

Reply:
Now I also think I understand your question about the parenthesis around {}.

You can also add a method, say IsZero(), to this struct, and then modify the conditional to

```
if foo.IsZero() {
    ...
}
```

Reply:
Fixed...